runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/FlinkKeyGroupStateInternals.java

```diff
@@ -71,8 +71,10 @@
   private final Map<String, Tuple2<Coder<?>, Map<String, ?>>>[] stateTables;
 
   public FlinkKeyGroupStateInternals(Coder<K> keyCoder, KeyedStateBackend keyedStateBackend) {
-    this.keyCoder = keyCoder;
-    this.keyedStateBackend = keyedStateBackend;
+    this.keyCoder = Preconditions.checkNotNull(keyCoder, "Coder for key must be provided.");
+    this.keyedStateBackend =
+        Preconditions.checkNotNull(
+            keyedStateBackend, "KeyedStateBackend must not be null. Missing keyBy call?");
```

Comment:
Note: Not strictly necessary but found it would improve the error message when a keyed stream was expected but keying had not been applied.